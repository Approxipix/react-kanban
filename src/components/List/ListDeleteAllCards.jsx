import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { deleteAllCards } from '../../redux/listReducer/actions/actions';
import { SubmitButton, Text } from '../BaseComponent';

class ListDeleteAllCards extends Component {
   deleteAllCards =() => {
    this.props.actions.deleteAllCards({
      listId: this.props.listId
    });
    this.props.toggleOpened();
  };

  render() {
    return (
      <>
        <Text>
          This will remove all the cards in this list from the board.
        </Text>
        <SubmitButton color={'danger'} onClick={() => this.deleteAllCards()}>
          Delete all
        </SubmitButton>
      </>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      deleteAllCards: deleteAllCards,
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(ListDeleteAllCards);
