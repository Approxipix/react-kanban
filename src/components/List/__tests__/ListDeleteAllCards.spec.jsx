import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import ListDeleteAllCards from '../ListDeleteAllCards';

const mockStore = configureStore([]);

describe('<ListDeleteAllCards>', () => {
  let store;
  let component;
  let mockToggleOpened = jest.fn();
  let mockDeleteCard = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();

    store = mockStore();
    store.dispatch = mockDeleteCard;
    const props = {
      listId: 'ListID',
      toggleOpened: mockToggleOpened,
    };

    const wrapper = shallow(<ListDeleteAllCards store={store} {...props} />);
    component = wrapper.find('ListDeleteAllCards').dive();
  });

  it('should render with given state from Redux store', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should dispatch an action after click on button", () => {
    const expectedAction = {
      type: 'DELETE_ALL_CARDS',
      payload: {
        listId: 'ListID',
      }
    };

    component.find('SubmitButton').simulate('click');
    expect(mockDeleteCard).toHaveBeenCalledTimes(1);
    expect(mockDeleteCard).toHaveBeenCalledWith(expectedAction);
  });

  it("should call toggleOpened after submit", () => {
    component.find('SubmitButton').simulate('click');

    expect(mockToggleOpened).toHaveBeenCalledTimes(1);
  });
});
