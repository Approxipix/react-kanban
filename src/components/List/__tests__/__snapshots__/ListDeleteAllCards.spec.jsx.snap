// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListDeleteAllCards> should render with given state from Redux store 1`] = `
<Fragment>
  <styled.p>
    This will remove all the cards in this list from the board.
  </styled.p>
  <SubmitButton
    color="danger"
    onClick={[Function]}
  >
    Delete all
  </SubmitButton>
</Fragment>
`;
